packages/squiggle-lang/src/rescript/Distributions/PointSetDist/PointSetDist_Scoring.res

```diff
@@ -1,46 +1,185 @@
+    let density = answer->estimatePdf
+    if density < 0.0 {
+      Operation.PdfInvalidError->Error
+    } else if density == 0.0 {
+      infinity->Ok
+    } else {
+      density->logFn->(x => -.x)->Ok
+    }
+  }
+  let scoreWithPrior' = (
+    ~estimatePdf: float => float,
+    ~answer: scalar,
+    ~priorPdf: float => float,
   ): result<float, Operation.Error.t> => {
-    let numerator = answer->predictionPdf
-    if numerator < 0.0 {
+    let numerator = answer->estimatePdf
+    let priorDensityOfAnswer = answer->priorPdf
+    if numerator < 0.0 || priorDensityOfAnswer < 0.0 {
       Operation.PdfInvalidError->Error
-    } else if numerator == 0.0 {
+    } else if numerator == 0.0 || priorDensityOfAnswer == 0.0 {
       infinity->Ok
     } else {
-      -.(
-        switch priorPdf {
-        | None => numerator->logFn
-        | Some(f) => {
-            let priorDensityOfAnswer = f(answer)
-            if priorDensityOfAnswer == 0.0 {
-              neg_infinity
-            } else {
-              (numerator /. priorDensityOfAnswer)->logFn
-            }
-          }
-        }
-      )->Ok
+      minusScaledLogOfQuot(~esti=numerator, ~answ=priorDensityOfAnswer)
     }
   }
+
+  let sum = (mp: PointSetTypes.MixedPoint.t): float => mp.continuous +. mp.discrete
+  let score = (~estimate: t, ~answer: scalar): result<float, Operation.Error.t> => {
+    let estimatePdf = x =>
+      switch estimate {
+      | Continuous(esti) => Continuous.T.xToY(x, esti)->sum
+      | Discrete(esti) => Discrete.T.xToY(x, esti)->sum
+      | Mixed(esti) => Mixed.T.xToY(x, esti)->sum
+      }
+
+    score'(~estimatePdf, ~answer)
+  }
+  let scoreWithPrior = (~estimate: t, ~answer: scalar, ~prior: t): result<
+    float,
+    Operation.Error.t,
+  > => {
+    let estimatePdf = x =>
+      switch estimate {
+      | Continuous(esti) => Continuous.T.xToY(x, esti)->sum
+      | Discrete(esti) => Discrete.T.xToY(x, esti)->sum
+      | Mixed(esti) => Mixed.T.xToY(x, esti)->sum
+      }
+    let priorPdf = x =>
+      switch prior {
+      | Continuous(prio) => Continuous.T.xToY(x, prio)->sum
+      | Discrete(prio) => Discrete.T.xToY(x, prio)->sum
+      | Mixed(prio) => Mixed.T.xToY(x, prio)->sum
+      }
+    scoreWithPrior'(~estimatePdf, ~answer, ~priorPdf)
+  }
 }
+
+module TwoScalars = {
+  let score = (~estimate: scalar, ~answer: scalar) =>
+    if answer == 0.0 {
+      0.0->Ok
+    } else if estimate == 0.0 {
+      infinity->Ok
+    } else {
+      minusScaledLogOfQuot(~esti=estimate, ~answ=answer)
+    }
+
+  let scoreWithPrior = (~estimate: float, ~answer: float, ~prior: float) =>
+    if answer == 0.0 {
+      0.0->Ok
+    } else if estimate == 0.0 || prior == 0.0 {
+      infinity->Ok
+    } else {
+      minusScaledLogOfQuot(~esti=estimate /. prior, ~answ=answer)
+    }
+}
+
+let logScore = (args: scoreArgs, ~combineFn, ~integrateFn, ~toMixedFn): result<
```

Comment:
I'd want to do some checking that the distributions are normalized

Reply:
https://github.com/quantified-uncertainty/squiggle/issues/719 resolving comment, new story filed.